daemon/graphdriver/btrfs/btrfs_test.go

```diff
@@ -40,25 +43,141 @@ func TestBtrfsSubvolDelete(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer d.Put("test")
 
-	if err := subvolCreate(dir, "subvoltest"); err != nil {
+	if err := subvolSnapshot("", dir, "subvoltest1"); err != nil {
 		t.Fatal(err)
 	}
 
-	if _, err := os.Stat(path.Join(dir, "subvoltest")); err != nil {
+	if _, err := os.Stat(path.Join(dir, "subvoltest1")); err != nil {
+		t.Fatal(err)
+	}
+
+	idir := path.Join(dir, "intermediate")
+	if err := os.Mkdir(idir, 0777); err != nil {
+		t.Fatalf("Failed to create intermediate dir %s: %v", idir, err)
+	}
+
+	if err := subvolSnapshot("", idir, "subvoltest2"); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := d.Put("test"); err != nil {
 		t.Fatal(err)
 	}
 
 	if err := d.Remove("test"); err != nil {
 		t.Fatal(err)
 	}
 
-	if _, err := os.Stat(path.Join(dir, "subvoltest")); !os.IsNotExist(err) {
+	if _, err := os.Stat(dir); !os.IsNotExist(err) {
 		t.Fatalf("expected not exist error on nested subvol, got: %v", err)
 	}
 }
 
+func TestBtrfsSubvolLongPath(t *testing.T) {
+	d := graphtest.GetDriver(t, "btrfs")
+	defer graphtest.PutDriver(t)
+
+	wdir, _ := os.Getwd()
```

Comment:
There is no excuse for ignoring the error here. The test should fail loudly, not silently fail to restore shared state which other tests in the package might rely upon.
```suggestion
	wdir, err := os.Getwd()
	if err != nil {
		t.Fatal(err)
	}
```